src/gui/folderman.cpp

```diff
@@ -307,6 +311,7 @@ void FolderMan::setupFoldersHelper(QSettings &settings, AccountStatePtr account,
             if (f) {
                 if (switchToVfs) {
                     f->switchToVirtualFiles();
+                    f->setRootPinState(PinState::OnlineOnly);
```

Comment:
@mgallien Isn't that what we need?

Reply:
so make it depends on `Theme::instance()->enforceVirtualFilesSyncFolder()` ?

Reply:
@mgallien No. That would not work. If I do this, then the PinState will be getting set on every app launch. But, we want it to be run just one time - when switching from non-VFS to VFS. Please check the implementation of this method `bool FolderMan::isSwitchToVfsNeeded(const FolderDefinition &folderDefinition) const` - it does check for the same macro `ENFORCE_VIRTUAL_FILES_SYNC_FOLDER`, plus, it checks if the VFS mode is not set already. This gives us the way to only set the pin state once.

Reply:
@mgallien Another problem with your idea. `bool FolderMan::isSwitchToVfsNeeded(const FolderDefinition &folderDefinition) const` depends on a specific folder. In case the folder is already switched (let's say we have multiple folders and some VFS and some are not), then, we must not touch that folder. `Theme::instance()->enforceVirtualFilesSyncFolder()` in contrast, does just check if the macro `ENFORCE_VIRTUAL_FILES_SYNC_FOLDER` is deined.

Reply:
but cannot you only do
```
if (switchToVfs) {
    f->switchToVirtualFiles();
    if (Theme::instance()->enforceVirtualFilesSyncFolder()) {
        f->setRootPinState(PinState::OnlineOnly);
    }
```

Reply:
@mgallien But, isn't `switchToVfs` already enough? It is not set to `true` if `ENFORCE_VIRTUAL_FILES_SYNC_FOLDER` is not set to `1`, so, calling `Theme::instance()->enforceVirtualFilesSyncFolder()` is redundant?

Reply:
the fact is that people that are switching from plain old sync folder have already the files data on their local storage
if you execute `f->setRootPinState(PinState::OnlineOnly);` and `Theme::instance()->enforceVirtualFilesSyncFolder()` is false will they keep their local storage data ?

Reply:
@mgallien Could you take a look at `bool FolderMan::isSwitchToVfsNeeded(const FolderDefinition &folderDefinition) const` implementation. It checks whether the option to enforce the VFS is on.

So, both my and your idea have drawbacks.